### PR TITLE
add support for pushover.net

### PR DESCRIPTION
**Pull request by floetenbaer**
add four new traccar.xml entries:

notificator.pushover.token  // (required) token from pushover.net
notificator.pushover.user // (required) user from pushover.net
notificator.pushover.device // optional: your user's device name to send the message directly to that device, rather than all of the user's devices (multiple devices may be separated by a comma)
notificator.pushover.title // optional: your message's title, otherwise your app's name is used

### CODE REVIEW

**Comment by floetenbaer** [comment]
comment was too long -> fixed

**Comment by tananaev** [comment]
Can each user configure their own devices?

**Comment by floetenbaer** [comment]
Yes. the ‚User‘ and the ‚key‘ are personalized values. So, the messages the massages are private.

**Comment by tananaev** [comment]
So, how each user can set it up?

**Comment by floetenbaer** [comment]
add the following in traccar.xml: (see https://pushover.net/api)

    <entry key='notificator.types'>web,mail,pushover</entry>
    (required) <entry key='notificator.pushover.token'>xxxx</entry>
    (required) <entry key='notificator.pushover.user'>xxxx</entry>
    (optional) <entry key='notificator.pushover.title'>demo.traccar.org</entry>
    (optional) <entry key='notificator.pushover.device'>myiPhone10, myiMac</entry>

**Comment by tananaev** [comment]
How would regular user access the config file? I think you misunderstood my question.

**Comment by floetenbaer** [comment]
No, the regular user can not access the properties. I did not figure out, how to change the properties in der web ui, like attributeMailSmtpHost. Do you have a documentation for that or a hint for me?

**Comment by tananaev** [comment]
You can look at some existing attributes for example.

**Comment by floetenbaer** [comment]
with pull request in traccar-web user can set the four conf. values, with overrides the traccar.xml settings

**Comment by tananaev** [comment]
I think now you went a little bit overboard with it. My idea was to replicate what we have for firebase. Basically only device token is configured per user. Sorry.

**Comment by floetenbaer** [comment]
No problem, tell me just, what you would accept. We have the pair user, key which identifies a account, normally a family or a small business. A account has one or more devices (normale a mobile phone or a computer), which are owned by a user. Fours we have a title of a message, which can be configured in the account by the account's admin. So I think: user and key goes in the traccar.xml, the device in the users properties and we drop the title. Whats your opinion?

**Comment by tananaev** [comment]
Yes, I think user and key goes into the config. Device(s) is configured per user. As for title, ideally it comes from the template.

**Comment by floetenbaer** [comment]
ok, finished (without title).

**Comment by tananaev** [comment]
It looks very similar to `NotificatorFirebase`. Can we unify the code? Also, it doesn't make sense to use `PropertiesProvider` for single attribute.

**Comment by floetenbaer** [comment]
Ok, done

File: src/main/java/org/traccar/notificators/NotificatorPushover.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 Anton Tananaev (anton@traccar.org)
```
Comment (tananaev) [inline src/main/java/org/traccar/notificators/NotificatorPushover.java]: Update the date here.

Comment (floetenbaer) [reply src/main/java/org/traccar/notificators/NotificatorPushover.java]: done

**Comment by floetenbaer** [comment]
but how to revert the submodule?

**Comment by tananaev** [comment]
Just revert it to the original commit.

**Comment by floetenbaer** [comment]
Im back on master branch in traccar-web but there is no change in the main repro to commit. So I don't know what to do :-(

**Comment by tananaev** [comment]
I don't even know how you managed to commit it because it's excluded. You likely have to temporary modify `.gitmodules` file.

**Comment by floetenbaer** [comment]
seems to be ok now :-)

**Comment by tananaev** [comment]
Merged, thanks.

**Comment by thenib** [comment]
Is it possible to add a parameter for pushovers critical alerts?

**Pull request merged**